hardhat/start.sh

```diff
@@ -32,4 +32,12 @@ nvm use 20 || fail "Error setting node version to v20"
 npm ci || fail "Error provisioning Hardhat"
 
 # Run Hardhat
-npx hardhat node --port 8545 || fail "Error running Hardhat"
\ No newline at end of file
+npx hardhat node --port 8545  &
+HARDHAT_NODE_PID=$!
+sleep 5 # Give the Hardhat node some time to start
+
+# Deploy contracts
+npx hardhat run scripts/deploy.js --network localhost || fail "Error deploying contracts"
```

Comment:
Oh yeah this is definitely a project-specific deployment, it should go in whichever project is leveraging the tests instead of OSHA.